Node-chakracore V8 build break on Ubuntu 16.04

To reproduce:
1. git clone the latest node-chakracore repo:
git clone https://github.com/nodejs/node-chakracore.git

2. cd to node-chakracore folder:
cd ~/GitHub/node-chakracore

3. make configuration  to v8 engine:
./configure --engine=v8

4. build:
make

We encountered following build break errors:

g++ '-DNODE_ENGINE="v8"' '-DNODE_ENGINE_V8' '-DNODE_ARCH="x64"' '-DNODE_PLATFORM="linux"' '-DNODE_WANT_INTERNALS=1' '-DV8_DEPRECATION_WARNINGS=1' '-DNODE_USE_V8_PLATFORM=1' '-DNODE_HAVE_I18N_SUPPORT=1' '-DNODE_HAVE_SMALL_ICU=1' '-DHAVE_INSPECTOR=1' '-DHAVE_OPENSSL=1' '-D__POSIX__' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=0' '-DUCONFIG_NO_BREAK_ITERATION=0' '-DHTTP_PARSER_STRICT=0' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-D_POSIX_C_SOURCE=200112' -I../src -I../tools/msvs/genfiles -I../deps/uv/src/ares -I/home/li2/GitHub/node-chakracore/out/Release/obj/gen -I/home/li2/GitHub/node-chakracore/out/Release/obj/gen/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common -I../deps/openssl/openssl/include -I../deps/zlib -I../deps/http_parser -I../deps/cares/include -I../deps/uv/include  -pthread -Wall -Wextra -Wno-unused-parameter -m64 -O3 -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++0x -MMD -MF /home/li2/GitHub/node-chakracore/out/Release/.deps//home/li2/GitHub/node-chakracore/out/Release/obj.target/node/src/async-wrap.o.d.raw   -c -o /home/li2/GitHub/node-chakracore/out/Release/obj.target/node/src/async-wrap.o ../src/async-wrap.cc
In file included from ../src/base-object-inl.h:29:0,
                 from ../src/async-wrap-inl.h:29,
                 from ../src/async-wrap.cc:23:
../src/env-inl.h: In constructor ‘node::Environment::AsyncHooks::AsyncHooks(v8::Isolate*)’:
../src/env-inl.h:87:7: error: class ‘node::Environment::AsyncHooks’ does not have any field named ‘uid_fields_ttdRef’
       uid_fields_ttdRef(nullptr) {
       ^
node.target.mk:197: recipe for target '/home/li2/GitHub/node-chakracore/out/Release/obj.target/node/src/async-wrap.o' failed
make[1]: *** [/home/li2/GitHub/node-chakracore/out/Release/obj.target/node/src/async-wrap.o] Error 1
rm 54c66d5aebf0b1cc95f8c1b6d449cc5ec5463317.intermediate
Makefile:76: recipe for target 'node' failed
make: *** [node] Error 2

Note: The output for "uname -a":
Linux li2-ubuntu 4.8.0-39-generic #42~16.04.1-Ubuntu SMP Mon Feb 20 15:06:07 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

--- a/src/env-inl.h
+++ b/src/env-inl.h
@@ -83,8 +83,11 @@ inline uint32_t* IsolateData::zero_fill_field() const {
 inline Environment::AsyncHooks::AsyncHooks(v8::Isolate* isolate)
     : isolate_(isolate),
       fields_(),
-      uid_fields_(),
-      uid_fields_ttdRef(nullptr) {
+      uid_fields_()
+#if ENABLE_TTD_NODE
+      ,uid_fields_ttdRef(nullptr)
+#endif
+{
   v8::HandleScope handle_scope(isolate_);
 
   // kAsyncUidCntr should start at 1 because that'll be the id the execution

--- a/test/message/nexttick_throw.v8.out
+++ b/test/message/nexttick_throw.v8.out
@@ -2,11 +2,10 @@
 *test*message*nexttick_throw.js:*
 *undefined_reference_error_maker;
 *^
-ReferenceError: *undefined_reference_error_maker*
+ReferenceError: *undefined_reference_error_maker is not defined*
 *at *test*message*nexttick_throw.js:*:*
 *at _combinedTickCallback (internal/process/next_tick.js:*:*)
 *at *_tickCallback (internal/process/next_tick.js:*:*)
 *at Module.runMain (module.js:*:*)
-*at run (bootstrap_node.js:*:*)
 *at startup (bootstrap_node.js:*:*)
 *at *bootstrap_node.js:*:*

--- a/test/parallel/test-vm-new-script-new-context.js
+++ b/test/parallel/test-vm-new-script-new-context.js
@@ -42,14 +42,16 @@ const Script = require('vm').Script;
 }
 
 {
-  assert.throws(
-    function() {
-      script.runInNewContext();
-    },
-    common.engineSpecificMessage({
-      v8: /^ReferenceError: foo is not defined$/,
-      chakracore: /ReferenceError: Use before declaration/
-    }));
+  const script = new Script('foo.bar = 5;');
+  assert.throws(function() {
+    script.runInNewContext();
+  }, common.engineSpecificMessage({
+    v8: /^ReferenceError: foo is not defined$/,
+    chakracore: /^TypeError: Unable to set property 'bar' of undefined or null reference$/
+  }));
+}
+
+{
   global.hello = 5;
   const script = new Script('hello = 2');
   script.runInNewContext();
@@ -95,14 +97,12 @@ const Script = require('vm').Script;
   script.runInNewContext({ f: f });
   assert.strictEqual(f.a, 2);
 
-  assert.throws(
-    function() {
-      script.runInNewContext();
-    },
-    common.engineSpecificMessage({
-      v8: /^ReferenceError: f is not defined$/,
-      chakracore: /'a' of undefined or null/
-    }));
+  assert.throws(function() {
+    script.runInNewContext();
+  }, common.engineSpecificMessage({
+    v8: /^ReferenceError: f is not defined$/,
+    chakracore: /^TypeError: Unable to set property 'a' of undefined or null reference$/
+  }));
 }
 
 {
@@ -111,6 +111,6 @@ const Script = require('vm').Script;
     script.runInNewContext.call('\'hello\';');
   }, common.engineSpecificMessage({
     v8: /^TypeError: this\.runInContext is not a function$/,
-    chakracore: /TypeError: Object doesn't support property or method 'runInContext'/
+    chakracore: /^TypeError: Object doesn't support property or method 'runInContext'$/
   }));
 }

